learning/courses/quantum-machine-learning/data-encoding.ipynb

```diff
@@ -0,0 +1,1713 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "7f428edb-ea96-48bb-adb2-aed91535dbeb",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<Image src=\"/learning/images/courses/quantum-machine-learning/data-encoding/extracted-outputs/7f428edb-ea96-48bb-adb2-aed91535dbeb-0.avif\" alt=\"Output of the previous code cell\" />"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Initializing parameters and empty lists for depths\n",
+    "x = [0.1, 0.2]\n",
+    "n_data = []\n",
+    "zz2gates = []\n",
+    "su22gates = []\n",
+    "z2gates = []\n",
+    "p2gates = []\n",
+    "\n",
+    "# Generating feature maps\n",
+    "for n in range(3, 10):\n",
+    "    x.append(n / 10)\n",
+    "    zzcircuit = ZZFeatureMap(n, reps=1, insert_barriers=True)\n",
+    "    zcircuit = ZFeatureMap(n, reps=1, insert_barriers=True)\n",
+    "    su2circuit = EfficientSU2(n, reps=1, insert_barriers=True)\n",
+    "    pcircuit = PauliFeatureMap(n, reps=1, paulis=[\"XX\"], insert_barriers=True)\n",
+    "    # Getting the cx depths\n",
+    "    zzcx = zzcircuit.decompose().count_ops().get(\"cx\")\n",
+    "    zcx = zcircuit.decompose().count_ops().get(\"cx\")\n",
+    "    su2cx = su2circuit.decompose().count_ops().get(\"cx\")\n",
+    "    pcx = pcircuit.decompose().count_ops().get(\"cx\")\n",
+    "\n",
+    "    # Appending the cx gate counts to the lists. We shift the zz & pauli data points, because they overlap.\n",
+    "    n_data.append(n)\n",
+    "    zz2gates.append(zzcx - 0.5)\n",
+    "    z2gates.append(0)\n",
+    "    su22gates.append(su2cx)\n",
+    "    p2gates.append(pcx + 0.5)\n",
+    "\n",
+    "# Plot the output\n",
+    "plt.plot(n_data, p2gates, \"bo\")\n",
+    "plt.plot(n_data, zz2gates, \"ro\")\n",
+    "plt.plot(n_data, su22gates, \"yo\")\n",
+    "plt.plot(n_data, z2gates, \"go\")\n",
+    "plt.ylabel(\"CX Gates\")\n",
+    "plt.xlabel(\"Data elements\")\n",
+    "plt.legend([\"Pauli\", \"ZZ\", \"SU2\", \"Z\"])\n",
+    "# plt.suptitle('ZZFeatureMap(n)')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "777bf066-af13-4065-a795-af19f67da099",
+   "metadata": {},
+   "source": [
+    "Generally Pauli and ZZ feature maps will result in greater circuit depth and higher numbers of 2-qubit gates than EfficientSU2 and Z feature maps.\n",
+    "\n",
+    "Because Qiskit's built-in feature maps are widely applicable, we will often not need to design our own, especially in the learning phase. However, experts in quantum machine learning will likely return to the subject of designing their own feature mapping, as they tackle two complicated challenges:\n",
```

Comment:
```suggestion
    "Because the feature maps built into Qiskit are widely applicable, we will often not need to design our own, especially in the learning phase. However, experts in quantum machine learning will likely return to the subject of designing their own feature mapping, as they tackle two complicated challenges:\n",
```

Reply:
We can't modify any trademarked terms in any way, such as "Qiskit's"